plots/laws/relativistic/photon_sail.py

```diff
@@ -0,0 +1,180 @@
+    energy_momentum_law.relativistic_energy,
+    incident_wave_energy,
+)
+
+total_momentum_before = incident_wave_momentum
+
+reflected_wave_momentum = electromagnetic_field_momentum_expr.subs(
+    energy_momentum_law.relativistic_energy,
+    reflected_wave_energy,
+)
+
+moving_mirror_momentum = moving_momentum_law.law.rhs.subs({
+    moving_momentum_law.rest_mass: mirror_rest_mass,
+    moving_momentum_law.velocity: mirror_speed,
+})
+
+# The reflected wave moves in the opposite direction
+total_momentum_after = moving_mirror_momentum - reflected_wave_momentum
+
+momentum_conservation_eqn = momentum_conservation_law.law.subs({
+    momentum_conservation_law.momentum(momentum_conservation_law.time_before): total_momentum_before,
+    momentum_conservation_law.momentum(momentum_conservation_law.time_after): total_momentum_after,
+})
+
+solved = solve(
+    (energy_conservation_eqn, momentum_conservation_eqn),
+    (mirror_speed, reflected_wave_energy),
+    dict=True,
+)[0]
+
+mirror_speed_expr = solved[mirror_speed]
+reflected_wave_energy_expr = solved[reflected_wave_energy]
+
+# Simplify the expressions by introducing a reduced energy symbol
+
+reduced_incident_wave_energy = Symbol("reduced_incident_wave_energy", positive=True)
+
+reduced_incident_wave_energy_eqn = Eq(
+    reduced_incident_wave_energy,
+    incident_wave_energy / (mirror_rest_mass * speed_of_light**2),
+)
+
+mirror_speed_expr = solve(
+    (Eq(mirror_speed, mirror_speed_expr), reduced_incident_wave_energy_eqn),
+    (mirror_speed, incident_wave_energy),
+    dict=True,
+)[0][mirror_speed]
+
+reflected_wave_energy_expr = solve(
+    (Eq(reflected_wave_energy, reflected_wave_energy_expr), reduced_incident_wave_energy_eqn),
+    (reflected_wave_energy, incident_wave_energy),
+    dict=True,
+)[0][reflected_wave_energy].replace(
+    4 * reduced_incident_wave_energy**2 + 4 * reduced_incident_wave_energy + 1,
+    (2 * reduced_incident_wave_energy + 1)**2,
+)
+
+# Print formulas
+
+print("Definition of reduced energy of incident wave:\n")
+print(print_expression(reduced_incident_wave_energy_eqn), end="\n\n\n")
+
+reduced_mirror_speed_expr = mirror_speed_expr / speed_of_light
+
+print("Formula of reduced speed of mirror:\n")
+print(print_expression(reduced_mirror_speed_expr), end="\n\n\n")
+
+reduced_reflected_wave_energy_expr = (
+    reflected_wave_energy_expr 
+    / (mirror_rest_mass * speed_of_light**2)
+)
+
+print("Formula of reduced energy of reflected wave:\n")
+print(print_expression(reduced_reflected_wave_energy_expr), end="\n\n\n")
+
+reduced_reflected_wave_energy_upper_limit = (
+    reduced_reflected_wave_energy_expr
+    .limit(reduced_incident_wave_energy, oo)
+)
+
+print("Upper limit of reduced energy of reflected wave:")
+print(print_expression(reduced_reflected_wave_energy_upper_limit))
+
+# Plot formulas
+
+base_plot = plot(
+    title=r"Mirror speed and reflected wave energy as functions of incident wave energy",
+    xlabel=r"$w_0 = \frac{W_0}{m_0 c^2}$",
```

Comment:
This legend is very cryptic. Try to add plain text, what parameters are on the plot.